custom_components/tuya_local/devices/rainpoint_water_timer.yaml

```diff
@@ -0,0 +1,116 @@
+name: RainPoint Smart Water Timer
+products:
+  - id: ew946yrp3pgbaziu
+    name: RainPoint TTV103FRF
+primary_entity:
+  entity: valve
+  class: water
+  icon: "mdi:watering-can"
+  dps:
+    - id: 108
+      type: boolean
+      name: valve
+      mapping:
+        - dps_val: true
+          value: false
+        - dps_val: false
+          value: true
+secondary_entities:
+  - entity: switch
+    name: Weekly Automatic Irrigation
+    category: config
+    icon: "mdi:calendar-clock"
+    dps:
+      - id: 125
+        optional: true
+        type: boolean
+        name: switch
+        mapping:
+          - dps_val: true
+            value: false
+          - dps_val: false
+            value: true
+  - entity: select
+    name: Weekly Irrigation Rain Delay
```

Comment:
This name is a bit long, other similar devices use just "Rain delay".   Also, naming needs to follow HA naming convention.. https://developers.home-assistant.io/docs/core/entity/#:~:text=Entity%20names%20should%20start%20with,a%20capitalized%20abbreviation%20of%20course).